src/v2/mod.rs

```diff
@@ -215,6 +215,100 @@ pub enum AccountIdentifier {
     Index(crate::types::AccountIndex),
 }
 
+/// Identifier for a relative epoch.
+#[derive(Debug, Copy, Clone)]
+pub struct RelativeEpoch {
+    /// Genesis index to query in.
+    pub genesis_index: types::GenesisIndex,
+    /// The epoch of the genesis to query.
+    pub epoch:         types::Epoch,
+}
+
+/// An identifier of an epoch used in queries.
+#[derive(Copy, Clone, Debug, derive_more::From)]
+pub enum EpochIdentifier {
+    /// A relative epoch to query.
+    Relative(RelativeEpoch),
+    /// Query the epoch of the block.
+    Block(BlockIdentifier),
+}
```

Comment:
I am not a big fan of using the term "relative" here. I would prefer "specified epoch". "Relative" should always be used in the context of what it's relative to (the genesis index), and we do not support "absolute" epoch queries.

Reply:
I would also prefer "Specified".